paddle/fluid/framework/data_transform.cc

```diff
@@ -33,11 +33,36 @@ void DataTransform(const OpKernelType& expected_kernel_type,
   Tensor in;
   in.ShareDataWith(input_tensor);
   Tensor out;
+  DataLayout lin = kernel_type_for_var.data_layout_;
+  DataLayout lout = expected_kernel_type.data_layout_;
 
   // do layout transform
-  if (NeedTransformLayout(expected_kernel_type.data_layout_,
-                          kernel_type_for_var.data_layout_)) {
-    TransDataLayout(kernel_type_for_var, expected_kernel_type, in, &out);
+  if (NeedTransformLayout(lout, lin)) {
+#ifdef PADDLE_WITH_MKLDNN
```

Comment:
Thanks @mozga-intel , I can understand that's a temporary solution. But I do no think the https://github.com/PaddlePaddle/Paddle/issues/10765 is trying to deal with this one. 

Maybe you can directly remove `#if` here since cudnn do not have global flag as well.

Reply:
@tensor-tang done.